### PR TITLE
Remove access to functions from inline assembly (unused code).

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by chriseth** [comment]
Note the assertion at the end: https://github.com/ethereum/solidity/pull/10209/files#diff-db8391050682bb61d15390fc5db2c0c5dcf3452d0c14fa4661b2c4a7725487a6R800

File: libsolidity/codegen/ContractCompiler.cpp [hunk 1]
```diff
@@ -691,26 +691,7 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
 		{
 			int const depositBefore = _assembly.stackHeight();
 			solAssert(!!decl->type(), "Type of declaration required but not yet determined.");
-			if (FunctionDefinition const* functionDef = dynamic_cast<FunctionDefinition const*>(decl))
```
Comment (axic) [inline libsolidity/codegen/ContractCompiler.cpp:694]: When was this last used? When did we export functions to assembly and when did we stop?

Comment (chriseth) [reply libsolidity/codegen/ContractCompiler.cpp:694]: some time ago.

Comment (axic) [reply libsolidity/codegen/ContractCompiler.cpp:694]: Reassuring.

**Review by axic — APPROVED** [review]

File: libsolidity/codegen/ContractCompiler.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/codegen/ContractCompiler.cpp:694]: Would really prefer to have an idea when we decided to do allow/disallow that.

**Comment by chriseth** [comment]
We properly disallowed it in 3e649eb8e1b85020de3aea52eb53e57d0c1743cb but I think the idea to disallow it was when we removed `jump`.

**Comment by chriseth** [comment]
Also relevant: https://github.com/ethereum/solidity/issues/8559

**Pull request merged**